internal/impl/qdrant/integration_test.go

```diff
@@ -0,0 +1,168 @@
+			payload: `{"content":"test payload"}`,
+		},
+		{
+			name:    "Test With multi vector",
+			pointID: `3`,
+			vector:  `{"some_multi": [[0.352,0.532,0.532],[0.352,0.532,0.532]]}`,
+			payload: `{"content":"test payload"}`,
+		},
+		{
+			name:    "Test With dense and sparse vector",
+			pointID: `"465213dd-3f11-4534-8daf-9fedf203549a"`,
+			vector:  `{"some_dense": [0.352,0.532,0.532],"some_sparse": {"indices": [23,325,532],"values": [0.352,0.532,0.532]}}`,
+			payload: `{"content":"test payload"}`,
+		},
+	}
+
+	containerPort, err := qdrantContainer.MappedPort(ctx, "6334/tcp")
+	if err != nil {
+		log.Fatalf("failed to get container port: %s", err)
+	}
+
+	setupCollection(ctx, fmt.Sprintf("localhost:%v", containerPort.Port()), collectionName)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			defaultQuery := func(ctx context.Context, testID, messageID string) (string, []string, error) {
+				return fmt.Sprintf(`{"content":"%v","id":%v}`, "hello world", messageID), nil, err
+			}
+
+			suite := integration.StreamTests(
+				// Is is possible to test output only without a `getFn GetMessageFunc` arg?
+				integration.StreamTestOutputOnlySendBatch(10, defaultQuery),
+			)
+			suite.Run(
+				t, template, integration.StreamTestOptPort(containerPort.Port()),
+				integration.StreamTestOptVarSet("POINT_ID", tc.pointID),
+				integration.StreamTestOptVarSet("COLLECTION_NAME", collectionName),
+				integration.StreamTestOptVarSet("VECTOR", tc.vector),
+				integration.StreamTestOptVarSet("PAYLOAD", tc.payload),
+			)
+		})
+	}
+
+	if err := qdrantContainer.Terminate(ctx); err != nil {
+		log.Fatalf("failed to terminate container: %s", err)
+	}
+}
+
+func setupCollection(ctx context.Context, host, collectionName string) {
+
+	conn, err := grpc.NewClient(host, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	if err != nil {
+		log.Fatalf("failed to instantiate client for tetsts: %s", err)
+	}
+
+	collectionsClient := pb.NewCollectionsClient(conn)
+
+	_, err = collectionsClient.Create(ctx, &pb.CreateCollection{
+		CollectionName: collectionName,
+		VectorsConfig: &pb.VectorsConfig{
+			Config: &pb.VectorsConfig_ParamsMap{
+				ParamsMap: &pb.VectorParamsMap{
+					Map: map[string]*pb.VectorParams{
+						// Default unnamed vector
+						// Created when using https://qdrant.tech/documentation/concepts/collections/#create-a-collection
+						"": {
+							Size:     3,
+							Distance: pb.Distance_Cosine,
+						},
+						"some_dense": {
+							Size:     3,
+							Distance: pb.Distance_Cosine,
+						},
+						"some_multi": {
+							Size:     3,
+							Distance: pb.Distance_Cosine,
+							MultivectorConfig: &pb.MultiVectorConfig{
+								Comparator: pb.MultiVectorComparator_MaxSim,
+							},
+						},
+					},
+				},
+			},
+		},
+		SparseVectorsConfig: &pb.SparseVectorConfig{
+			Map: map[string]*pb.SparseVectorParams{
+				"some_sparse": {},
+			},
+		},
+	})
+
+	if err != nil {
```

Comment:
Can we have this helper return an error and using `require.NoError` in the test itself?